src/stk/molecular/topology_graphs/polymer/linear/linear.py

```diff
@@ -587,9 +594,35 @@ def _get_building_block_vertices(self, building_blocks, vertices):
                 building_block_vertices.get(bb, [])
             )
             building_block_vertices[bb].append(vertex)
+
+        building_block_vertices = self._with_unaligning_vertices(
+            building_block_vertices=building_block_vertices,
+        )
         return building_block_vertices
 
-    def _get_scale(self, building_block_vertices):
+    @staticmethod
+    def _with_unaligning_vertices(
+        building_block_vertices: dict[BuildingBlock, type[Vertex]],
+    ) -> dict[BuildingBlock, type[Vertex]]:
+        clone = dict(building_block_vertices)
```

Comment:
you don't haveto clone it, just make an empty dict and then fill it in the for loop